app/support/products/scheduling_support.rb

```diff
@@ -155,14 +155,19 @@ class ReservationFinder
     attr_accessor :time, :rule, :day_start, :day_end, :options
 
     def initialize(time, rule, options = {})
-      self.time = time
+      self.time = rule.instrument.cutoff_hours > 0 ? time_with_cutoff_hours(time, rule.instrument) : time
       self.rule = rule
       self.day_start = Time.zone.local(time.year, time.month, time.day, rule.start_hour, rule.start_min, 0)
       self.day_end   = Time.zone.local(time.year, time.month, time.day, rule.end_hour, rule.end_min, 0)
       self.options   = options
       adjust_time
     end
 
+    def time_with_cutoff_hours(time, instrument)
+      first_available_time = Time.zone.now + instrument.cutoff_hours.hours
+      time < first_available_time ? first_available_time : time
```

Comment:
What about one-lining it to something like `[time, instrument.cutuff_hours.hours.from_now].max`. Then we might not even need the extracted method.